File: C#/Monotonic Stack/739. Daily Temperatures.cs
/* 739. Daily Temperatures 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/daily-temperatures/submissions/1384488121/ */

public class Solution {
    public int[] DailyTemperatures(int[] temps) {
        int[] results = new int[temps.Length];
        Stack<int> stack = new Stack<int>();
        
        for (int i = 0; i < temps.Length; i++) {
            while (stack.Count > 0 && temps[stack.Peek()] < temps[i]) {
                int index = stack.Pop();
                results[index] = i - index;
            }
            stack.Push(i);
        }

        return results;
    }
}

File: Python/Monotonic Stack/739. Daily Temperatures.py
""" 739. Daily Temperatures 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/daily-temperatures/submissions/1384487086/ """

class Solution:
    def dailyTemperatures(self, temps):
        
        results = [0] * len(temps)
        stack = []

        for i, temp in enumerate(temps):

            while stack and temps[stack[-1]] < temp:

                index = stack.pop()
                results[index] = i - index

            stack.append(i)

        return results